### ISSUE TITLE
DSIFN-CD - Not able to reproduce results

### ISSUE THREAD

**Issue by zaidbhat1234** [wgcban/ChangeFormer#51]
Hi @wgcban, 
I am unable to reproduce your results on DSIFN dataset. I was able to reproduce them on the LEVIR-CD dataset. The results I get are closer to BIT on DSIFN but nowhere close to the ones you reported. I use the default script provided to run DSIFN training and also downloaded the processed dataset given by you. I read in other issues that you might have an issue with same images in train/test in some cases which led to higher accuracy for you. Could you please verify if you are still able to reproduce your results?
I see that you have used a different learning rate for DSIFN. Is that correct or do I need to use the same training config for both as reported in the paper?

**Comment by wgcban** [comment]
@zaidbhat1234 As shown in the training/validation/testing logs, the reported values in the paper match exactly with the results I obtained for the DSIFN test splits. I apologize for the discrepancy in the learning rates mentioned in the paper and the code. However, I can confirm that the learning rate specified in the SH file is the one I used for the DSIFN dataset experiments. Could you confirm if you followed the exact dataset preparation steps that I used for the experiments? Thank you.

**Comment by wgcban** [comment]
Here is the log file: https://www.dropbox.com/sh/yrdbmkvhbpxuabw/AACRiX_kI5s2Dpd90sCtTUUJa/CD_ChangeFormerV6_DSIFN_b16_lr0.00006_adamw_train_test_200_linear_ce_multi_train_True_multi_infer_False_shuffle_AB_False_embed_dim_256?dl=0&preview=log.txt&subfolder_nav_tracking=1

**Comment by wgcban** [comment]
@zaidbhat1234 At the same time, you can use the provided checkpoint to evaluate the accuracy of the model on DSIFN test split.

**Comment by zaidbhat1234** [comment]
@wgcban Thanks for your comments. The training loss was oscillating initially but around 100 epochs I could see some good results and the training stabilize. I downloaded the dataset from the link that you shared in the ReadMe. I am guessing that dataset is already prepared and ready to use right?

**Comment by zaidbhat1234** [comment]
@wgcban I also had another question:
1. Did you use the same train/val/test split for ChangeFormer when you compare its performance with other CD models like BIT?
2. I notice that majority of images used in test.txt are also present in train.txt (164/192) ~85%. Is this something that is an error or it was the same in previous CD works such as BIT? Could this be the reason for high difference in metrics from previous works on DSIFN?
